server/services/task.queue.ts

```diff
@@ -0,0 +1,41 @@
+import { IChunkScheme, Chunk, Status } from '../models/chuck.model';
+import { processChunk } from './process-chunk.service';
+
+export class TaskQueue {
```

Comment:
TaskQueue must be a singleton otherwise we will have many instances of it trying to do maybe the same job